src/pages/api/fake/dataset/[id].ts

```diff
@@ -0,0 +1,20 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import type { Dataset } from '../../../dataspace/[dataspace-id]/dataset/[dataset-id]'
+import datasets from './datasets.json'
+
+// eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+const handler = (req: NextApiRequest, res: NextApiResponse): void => {
+  if (req.method !== 'GET') res.status(404).send('Not found')
+
+  const id = req.query.id as string
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const dataset: unknown | undefined = datasets.find((item: any) => item.id === id)
```

Comment:
Unnecessary type declaration, you can let the TS engine infer the type.

Reply:
Without this type declaration, we have the linter `Unexpected any. Specify a different type.eslint[@typescript-eslint/no-explicit-any](https://typescript-eslint.io/rules/no-explicit-any)` warning.

If we use `unknown` instead, we can't do `item.id`.

Reply:
If the engine can't infer the type, it's because one or many types used are already `any`...
Why `item` is `any`?

Reply:
Here item is any because we use a json to instanciate `datasets`.

Reply:
Yes, as suggested by @fredericvilcot, introduce a type to define a 'Dataset'. And since you've already defined this type in the frontend, use the ability of Next.js to share this type (and all those involved in the APIs) between the frontend and the backend